### PR TITLE
setDuration member function with unit test

### PR DESCRIPTION
**Pull request by jmborr**
**References** [Update run's duration when TimeROI's updated](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=948)

**Description of work.**

- [x] `Run::setDuration( )` member function with unit test

**To test:**

Just code review.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by bolotovskyr — DISMISSED** [review]
Looks straightforward.

File: Framework/Algorithms/test/FilterByLogValueTest.h [hunk 1]
```diff
@@ -112,8 +112,8 @@ class FilterByLogValueTest : public CxxTest::TestSuite {
                                                   // WorkspaceCreationHelper.
 
     TimeSeriesProperty<double> *temp;
-    temp = new TimeSeriesProperty<double>("temp");
-    // 10 C at 10 sec up to 50C at 50 sec
+    temp = new TimeSeriesProperty<double>("temp"); // Temperature log
+    // 10C at 10 sec up to 50C at 50 seconds
```
Comment (bolotovskyr) [inline Framework/Algorithms/test/FilterByLogValueTest.h:116]: Just a minor thing: in the comment, either use "sec" or "seconds" in both cases.

File: Framework/API/src/Run.cpp [hunk 2]
```diff
@@ -368,7 +369,27 @@ void Run::integrateProtonCharge(const std::string &logname) const {
...
 
 //-----------------------------------------------------------------------------------------------
+/**
+ * Update property "duration" with the duration of the Run's TimeROI attribute.
+ *
+ * If the Run's TimeROI is empty, this member function does nothing.
+ */
+void Run::setDuration() {
+  if (m_timeroi->empty())
+    return;
```
[the remark this answers is not in the document]
Comment (jmborr) [reply Framework/API/src/Run.cpp:379]: @bolotovskyr  `setDuration()` is meant to be called when updating the `TimeROI` of the run object, which is why I decided to do nothing if there's no `TimeROI` (a.k.a the `TimeROI` is empty). This is the least intrusive option. Other option would be to find out if there's a `run_start` and `end_start` and compute a "duration" from these values. Other option is to set the duration to `zero`. The point is that there's not a uniquely defined way when one can't make use of the `TimeROI`.

**Review by bolotovskyr — APPROVED** [review]
Good to go.

**Pull request merged**